vnext/JSI/Desktop/ChakraJsiRuntime_core.cpp

```diff
@@ -219,65 +139,51 @@ void ChakraRuntime::setupNativePromiseContinuation() noexcept {
-  wchar_t key[_countof(L"VS_VERSION_INFO")];
-  uint16_t padding1;
-  VS_FIXEDFILEINFO fixedFileInfo;
-  uint32_t padding2;
-};
-#endif
-
-// TODO :: This code is mostly copied from the old ChakraExecutor flow, and not
-// verified for reliability yet.
-// TODO :: Re-evaluate this strategy of finding the dll version for versioning
-// the runtime.
-/*static*/ void ChakraRuntime::initRuntimeVersion() noexcept {
-  // This code is win32 only at the moment. We will need to change this
-  // line if we want to support UWP.
-#if !defined(CHAKRACORE_UWP)
-  constexpr wchar_t chakraDllName[] = L"ChakraCore.dll";
+void ChakraRuntime::startDebuggingIfNeeded() {
+  auto &args = runtimeArgs();
+  if (args.enableDebugging) {
+    auto port =
+        args.debuggerPort == 0 ? DebuggerDefaultPort : args.debuggerPort;
+    auto runtimeName = args.debuggerRuntimeName.empty()
+        ? DebuggerDefaultRuntimeName
+        : args.debuggerRuntimeName;
 
-  auto freeLibraryWrapper = [](void *p) { FreeLibrary((HMODULE)p); };
-  HMODULE moduleHandle;
-  if (!GetModuleHandleExW(0, chakraDllName, &moduleHandle))
-    return;
+    JsErrorCode result = enableDebugging(
+        m_runtime,
+        runtimeName,
+        args.debuggerBreakOnNextLine,
+        static_cast<uint16_t>(port),
+        m_debugProtocolHandler,
+        m_debugService);
 
-  std::unique_ptr<void, decltype(freeLibraryWrapper)> moduleHandleWrapper{
-      moduleHandle, std::move(freeLibraryWrapper)};
+    if (result == JsNoError) {
+      m_debugPort = port;
+      m_debugRuntimeName = runtimeName;
+    }
+  }
 
-  HRSRC versionResourceHandle =
-      FindResourceW(moduleHandle, MAKEINTRESOURCE(VS_VERSION_INFO), RT_VERSION);
+  if (args.debuggerBreakOnNextLine && m_debugProtocolHandler) {
+    if (args.loggingCallback)
+      args.loggingCallback(
+          "Waiting for debugger to connect...", LogLevel::Info);
 
-  if (!versionResourceHandle ||
-      SizeofResource(
-          static_cast<HMODULE>(moduleHandleWrapper.get()),
-          versionResourceHandle) < sizeof(FileVersionInfoResource))
-    return;
+    m_debugProtocolHandler->WaitForDebugger();
 
-  HGLOBAL versionResourcePtrHandle =
-      LoadResource(moduleHandle, versionResourceHandle);
-  if (!versionResourcePtrHandle)
-    return;
+    if (args.loggingCallback)
+      args.loggingCallback("Debugger connected", LogLevel::Info);
+  }
+}
 
-  FileVersionInfoResource *chakraVersionInfo =
-      static_cast<FileVersionInfoResource *>(
-          LockResource(versionResourcePtrHandle));
-  if (!chakraVersionInfo)
-    return;
+void ChakraRuntime::stopDebuggingIfNeeded() {
+  if (m_debugService) {
+    JsErrorCode result = m_debugService->Close();
 
-  s_runtimeVersion = chakraVersionInfo->fixedFileInfo.dwFileVersionMS;
-  s_runtimeVersion <<= 32;
-  s_runtimeVersion |= chakraVersionInfo->fixedFileInfo.dwFileVersionLS;
-#endif
+    if (result == JsNoError) {
+      result = m_debugService->UnregisterHandler(m_debugRuntimeName);
+    }
+  }
```

Comment:
>} [](start = 2, length = 1)

(nit) add empty line after brace

Reply:
I can't because of clang format... :(

---
In reply to: [333112847](https://github.com/microsoft/react-native-windows/pull/3366#discussion_r333112847) [](ancestors = 333112847)